backend/src/apiserver/template/v2_template.go

```diff
@@ -20,7 +35,7 @@ type V2Spec struct {
 	spec *pipelinespec.PipelineSpec
 }
 
-func (t *V2Spec) ScheduledWorkflow(apiJob *api.Job) (*scheduledworkflow.ScheduledWorkflow, error) {
+func (t *V2Spec) ScheduledWorkflow(inputInterface interface{}) (*scheduledworkflow.ScheduledWorkflow, error) {
```

Comment:
Maybe I missed something. This entire file should be v2 only, why do we need to handle both v1 and v2 APIs here?

Reply:
The "v2" in the file name `v2_template.go` refers to the IR version. It used to handle v1 api, so it now handles both v1 and v2 apis.

Reply:
I realized maybe we haven't discussed this design choice clearly--should v2 APIs only support v2 behaviors? if not, should all v2 APIs support v1 behaviors as well, or only part of them need to do so?

My initial thought on this: All the create APIs don't need to handle v1 cases. More specifically, if I have an Argo workflow yaml, should I be able to create a run/recurring run using it via v2 APIs? Answer should be no, right?